pkg/rancher-desktop/preload/extensions.ts

```diff
@@ -334,27 +334,73 @@ class Client implements v1.DockerDesktopClient {
   docker = {
     cli:            { exec: getExec('docker-cli') },
     listContainers: async(options: DockerListContainersOptions = {}) => {
-      const args = ['ls', '--format={{json .}}', '--no-trunc'];
+      // Unfortunately, there's no command line option to just make an API call,
+      // and `container ls` by itself doesn't provide all the info.
+      const lsArgs = ['ls', '--format={{json .}}', '--no-trunc'];
 
-      args.push(`--all=${ options.all ?? false }`);
+      lsArgs.push(`--all=${ options.all ?? false }`);
       if ((options.limit ?? -1) > -1) {
-        args.push(`--last=${ options.limit }`);
+        lsArgs.push(`--last=${ options.limit }`);
       }
-      args.push(`--size=${ options.size ?? false }`);
       if (options.filters !== undefined) {
-        args.push(`--filter=${ options.filters }`);
+        lsArgs.push(`--filter=${ options.filters }`);
       }
       if (options.namespace) {
-        args.unshift(`--namespace=${ options.namespace }`);
+        lsArgs.unshift(`--namespace=${ options.namespace }`);
       }
 
-      const result = await this.docker.cli.exec('container', args);
+      const lsResult = await this.docker.cli.exec('container', lsArgs);
 
-      if (result.code || result.signal) {
-        throw new Error(`failed to list containers: ${ result.stderr }`);
+      if (lsResult.code || lsResult.signal) {
+        throw new Error(`failed to list containers: ${ lsResult.stderr }`);
       }
 
-      return result.parseJsonLines();
+      const lsContainers = lsResult.parseJsonLines();
+
+      // We need to run `container inspect` to add more info.
+      const inspectArgs = ['--format={{json .}}'];
+
+      inspectArgs.push(`--size=${ options.size ?? false }`);
+      inspectArgs.push(...lsContainers.map(c => c.ID));
+      if (options.namespace) {
+        inspectArgs.unshift(`--namespace=${ options.namespace }`);
+      }
+
+      const inspectResults = await this.docker.cli.exec('inspect', inspectArgs);
+
+      if (inspectResults.code || inspectResults.signal) {
+        throw new Error(`failed to inspect containers: ${ inspectResults.stderr }`);
+      }
+
+      const inspectContainers = inspectResults.parseJsonLines().flat();
+
+      return lsContainers.map((c) => {
+        const details = inspectContainers.find(i => i.Id.startsWith(c.ID));
+        const pick = (object: any, ...prop: (string | [string, string])[]) => {
```

Comment:
We're making our own `pick` here because we can't pull in any NodeJS packages (e.g. lodash) from the preload context; it breaks loading the script.